src/pip/_vendor/tenacity/__init__.py

```diff
@@ -23,7 +23,9 @@
     iscoroutinefunction = None
 
 try:
-    import tornado
+    # Pip does not currently vendor Tornado, so this will take the
+    # ImportError path.
+    from pip._vendor import tornado
```

Comment:
I think it would be better to patch the entire block to `tornado = None`. If the import always fails, we don't need to try that.

Reply:
I'm happy to make that change. It was my original plan, but I decided to use `from pip._vendor import tornado` so that in the (unlikely) event pip adds Tornado to its vendored dependencies things will just work and you won't be left with a copy of tenacity that mostly works but breaks if it interacts with tornado.